lib/src/tasks/analyze/cli.dart

```diff
@@ -41,10 +45,22 @@ class AnalyzeCli extends TaskCli {
     bool fatalWarnings = TaskCli.valueOf(
         'fatal-warnings', parsedArgs, config.analyze.fatalWarnings);
     bool hints = TaskCli.valueOf('hints', parsedArgs, config.analyze.hints);
+    bool fatalHints =
+        TaskCli.valueOf('fatal-hints', parsedArgs, config.analyze.failOnHints);
+
+    if (!hints && fatalHints) {
```

Comment:
I'm torn on this. I think I like that idea, but it could be confusing to make `--hints` function differently than the dartanalyzer.